pkg/obfuscate/cache.go

```diff
@@ -51,28 +51,26 @@ func (c *measuredCache) statsLoop() {
 }
 
 type cacheOptions struct {
-	On     bool
-	Statsd StatsClient
+	On      bool
+	Statsd  StatsClient
+	MaxSize int64
 }
 
+const worstCaseQuerySize = 11
+
 // newMeasuredCache returns a new measuredCache.
 func newMeasuredCache(opts cacheOptions) *measuredCache {
 	if !opts.On {
 		// a nil *ristretto.Cache is a no-op cache
 		return &measuredCache{}
 	}
 	cfg := &ristretto.Config{
-		// We know that the maximum allowed resource length is 5K. This means that
-		// in 5MB we can store a minimum of 1000 queries.
-		MaxCost: 5000000,
-
+		MaxCost: opts.MaxSize,
 		// An appromixated worst-case scenario when the cache is filled with small
-		// queries averaged as being of length 11 ("LOCK TABLES"), we would be able
-		// to fit 476K of them into 5MB of cost.
-		//
-		// We average it to 500K and multiply 10x as the documentation recommends.
-		NumCounters: 500000 * 10,
-
+		// queries averaged as being of length 11 ("LOCK TABLES").
+		// opts.MaxSize / worstCaseQuerySize would be the maximum number of queries.
+		// We multiply it by 10 as per the recommendation in the ristretto documentation.
+		NumCounters: opts.MaxSize / worstCaseQuerySize * 10,
```

Comment:
`MaxSize` is 5,000,000 bytes by default.
The worst-case query size could be 11 bytes = `len([]byte("LOCK TABLES"))`.
454,545 items (= 5,000,000 bytes / 11 bytes) should be the maximum number of items in the worst-case scenario.
[As per the recommendation by the library](https://github.com/outcaste-io/ristretto/blob/v0.2.3/cache.go#L83-L94), 4,545,450 (= 454,545 items * 10) should be set in `NumCounters`.

The value 4,545,450 is smaller than the value that was originally hardcoded in `NumCounters` (= 500000 * 10).